File: testdata/java_sources/Cache.java
package com.example;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * Simple LRU cache implementation.
 */
public class Cache<K, V> {

    private final int capacity;
    private final Map<K, V> map;

    public Cache(int capacity) {
        this.capacity = capacity;
        this.map = new LinkedHashMap<>(capacity, 0.75f, true) {
            protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
                return size() > Cache.this.capacity;
            }
        };
    }

    public synchronized V get(K key) {
        return map.get(key);
    }

    public synchronized void put(K key, V value) {
        map.put(key, value);
    }

    public synchronized int size() {
        return map.size();
    }

    /**
     * Remove a key from the cache - dead code.
     */
    public synchronized void remove(K key) {
        map.remove(key);
    }

    /**
     * Clear all entries - dead code.
     */
    public synchronized void clear() {
        map.clear();
    }

    /**
     * Check if key exists - dead code.
     */
    public synchronized boolean containsKey(K key) {
        return map.containsKey(key);
    }
}

File: testdata/kotlin_sources/Cache.kt
package com.example.kotlin

import java.util.concurrent.ConcurrentHashMap

/**
 * Thread-safe cache implementation.
 */
class KotlinCache<K, V>(private val maxSize: Int = 1000) {

    private val map = ConcurrentHashMap<K, V>()

    fun put(key: K, value: V): V? {
        if (map.size >= maxSize) {
            // Evict first entry (simplified LRU)
            map.remove(map.keys.first())
        }
        return map.put(key, value)
    }

    fun get(key: K): V? = map[key]

    fun remove(key: K): V? = map.remove(key)

    fun size(): Int = map.size

    /**
     * Clear all entries - dead code.
     */
    fun clear() = map.clear()

    /**
     * Get or compute - dead code.
     */
    fun getOrPut(key: K, compute: () -> V): V {
        return map.getOrPut(key, compute)
    }

    /**
     * Contains check - dead code.
     */
    fun contains(key: K): Boolean = map.containsKey(key)
}

/**
 * Unused global cache instance - dead code.
 */
val globalCache = KotlinCache<String, Any>()
